test/rpc/test_rpc_transactions.py

```diff
@@ -607,3 +689,115 @@ def test_add_deploy_transaction(deploy_content, version):
 
     assert is_felt(receipt["transaction_hash"])
     assert is_felt(receipt["contract_address"])
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "private_key, public_key, account_salt",
+    [
+        (
+            0x6F9E0F15B20753CE2E2B740B182099C4ADF765D0C5A5B75C1AF3327358FBF2E,
+            0x7707342F75277F32F1A0AD532E1A12016B36A3967332D31F915C889678B3DB6,
+            0x75B567ECB69C6D032982FA32C8F52D2F00DB50C5DE2C93EDDA70DE9B5109F8F,
+        )
+    ],
+)
+def test_add_deploy_account_transaction_on_incorrect_class_hash(
+    private_key, public_key, account_salt
+):
+    """
+    Add deploy transaction on incorrect class
+    """
+    invalid_class_hash = 1337
+
+    deploy_account_tx, address = _prepare_deploy_account_tx(
+        private_key, public_key, account_salt, oz_account_class
+    )
+    rpc_deploy_account_tx = _rpc_deploy_account_from_gateway(deploy_account_tx)
+    rpc_deploy_account_tx["class_hash"] = rpc_felt(invalid_class_hash)
+
+    mint(hex(address), amount=int(1e18))
+
+    ex = rpc_call(
+        "starknet_addDeployAccountTransaction",
+        params={"deploy_account_transaction": rpc_deploy_account_tx},
+    )
+    assert ex["error"] == {"code": 28, "message": "Class hash not found"}
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
```

Comment:
Why did you parametrize this test when there's only one case? The next test case as well.

Reply:
I thought it would make the test clearer. I can move them back inside the tests though.